.github/workflows/publish_canary.yml

```
@@ -0,0 +1,42 @@
+name: Publish Canary to NPM
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Branch to release from"
+        required: true
+        default: "v2.0"
+      canaryVersion:
+        description: "Custom version (e.g., 1.2.3-canary.42)"
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch || github.ref_name }}
```

```suggestion
          ref: ${{ github.event.inputs.branch  }}
```
removing the condition